### PR TITLE
Performance Improvements

### PR DESCRIPTION
**Pull request by yorek**
Using JSON instead of Row Constructors
Reduced active connection usage 
Using more realistic test table

### CODE REVIEW

File: src/SqlBinding/SqlAsyncCollector.cs [hunk 1]
```diff
@@ -379,8 +381,55 @@ public static TableInformation RetrieveTableInformation(SqlConnection sqlConnect
                     primaryKeyMatchingQuery += $" AND ExistingData.{primaryKey} = NewData.{primaryKey}";
                 }
 
+                // 2. Get Columns Definition
+                string columnDefinitionQuery = $@"
```
Comment (glennamanns) [inline src/SqlBinding/SqlAsyncCollector.cs:385]: what benefits does grabbing the column types offer? More performance? or is it just required to use the JSON functionality?

**Review by glennamanns — APPROVED** [review]
thanks for making these changes, Davide! I will merge them into my branch and work on the additional changes necessary (refactoring, support different orders of columns, etc.)

File: src/SqlBinding/SqlAsyncCollector.cs [hunk 2]
```diff
@@ -319,44 +300,61 @@ public class TableInformation
...
-                if (!string.IsNullOrEmpty(schema))
-                {
-                    primaryKeyQuery += $"AND Col.Constraint_Schema = '{schema}'";
-                }
+                IEnumerable<string> columnNames = typeof(T).GetMembers().Where(f => f.MemberType == MemberTypes.Property || f.MemberType == MemberTypes.Field).Select(f => f.Name);
 
                 var primaryKeys = new List<string>();
+                var columnsDefinition = new List<string>();
+
+                // 1. Query SQL to get primary keys
```
[the remark this answers is not in the document]
Comment (yorek) [reply src/SqlBinding/SqlAsyncCollector.cs:333]: 300 ms are needed just the first time, as connection needs to be established...that's what is taking so much time. After that, connection is pooled and time will be something like 2 msec.

[the remark this answers is not in the document]
Comment (yorek) [reply src/SqlBinding/SqlAsyncCollector.cs:333]: Btw, query is fine, I just cleaned up it a bit

File: src/SqlBinding/SqlAsyncCollector.cs (hunk 1 above)
Comment (yorek) [reply src/SqlBinding/SqlAsyncCollector.cs:385]: It is required for using OPENJSON

Comment (yorek) [reply src/SqlBinding/SqlAsyncCollector.cs:385]: https://docs.microsoft.com/en-us/sql/relational-databases/json/use-openjson-with-an-explicit-schema-sql-server?view=sql-server-ver15

**Pull request merged**